### PR TITLE
Warn when specifying resource limits in hosted Vespa

### PR DESCRIPTION
**Pull request by hmusum**
Want to throw and fail deployments, but need to make sure nobody
is doing this first, so warn for now.

### CODE REVIEW

File: config-model/src/test/java/com/yahoo/vespa/model/content/ClusterResourceLimitsTest.java [hunk 1]
```diff
@@ -114,6 +119,29 @@ public void default_resource_limits_when_feed_block_is_enabled_in_distributor()
                 new Fixture(true));
     }
 
+    @Test
+    // TODO: Change to expect exception being thrown when noone uses this in hosted
```
Comment (geirst) [inline config-model/src/test/java/com/yahoo/vespa/model/content/ClusterResourceLimitsTest.java:123]:
```suggestion
    // TODO: Change to expect exception being thrown when no one uses this in hosted
```

**Review by geirst — APPROVED** [review]
👍

**Comment by baldersheim** [comment]
Are we doing the same for both proton resources and cluster resources ?

**Comment by geirst** [comment]
Yes, as DomResourceLimitsBuilder (with the check for hosted flag) is used for both cluster wide and proton limits in ClusterResourceLimits.

File: config-model/src/main/java/com/yahoo/vespa/model/content/cluster/DomResourceLimitsBuilder.java [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
 
-    public static ResourceLimits.Builder createBuilder(ModelElement contentXml) {
+    public static ResourceLimits.Builder createBuilder(ModelElement contentXml, boolean hostedVespa, DeployLogger deployLogger) {
         ResourceLimits.Builder builder = new ResourceLimits.Builder();
         ModelElement resourceLimits = contentXml.child("resource-limits");
-        if (resourceLimits == null) {
+        if (resourceLimits == null) { return builder; }
+
+        if (hostedVespa) {
+            deployLogger.log(Level.WARNING, "Element " + resourceLimits + " is not allowed, default limits will be used");
```
Comment (freva) [inline config-model/src/main/java/com/yahoo/vespa/model/content/cluster/DomResourceLimitsBuilder.java:23]: Log with `.logApplicationPackage()` so the tenant gets a notification in console?

Comment (hmusum) [reply config-model/src/main/java/com/yahoo/vespa/model/content/cluster/DomResourceLimitsBuilder.java:23]: Yes, thanks, done in https://github.com/vespa-engine/vespa/pull/17948

**Pull request merged**